delta-lake/common/src/main/scala/com/nvidia/spark/rapids/delta/RapidsDeltaWrite.scala

```diff
@@ -68,12 +68,29 @@ case class RapidsDeltaWrite(child: LogicalPlan) extends UnaryNode {
  * a GpuRapidsDeltaWriteExec during GPU planning and should never be executed.
  */
 case class RapidsDeltaWriteExec(child: SparkPlan) extends V2CommandExec with UnaryExecNode {
+  @transient lazy val doExecuteMethod = {
+    val doExecute = classOf[SparkPlan].getDeclaredMethod("doExecute")
+    doExecute.setAccessible(true)
+    doExecute
+  }
+
   override def output: Seq[Attribute] = child.output
 
   override def run(): Seq[InternalRow] = {
     throw new IllegalStateException("Should have been replaced with a GPU node before execution")
   }
 
+  override protected def doExecute(): RDD[InternalRow] = {
+    // During execution of Delta Lake writes, we sometimes make the entire plan
+    // fall back to CPU due to the query being categorized as a Delta Lake
+    // Metadata Query, such as when there are references to internal
+    // metadata fields in a Parquet scan. This leaves the
+    // RapidDeltaWriteExec in the plan, so we need to support row-based
+    // execution in this case.
+    logWarning("RapidsDeltaWriteExec performing row-based execution")
+    doExecuteMethod.invoke(child).asInstanceOf[RDD[InternalRow]]
```

Comment:
This seems to be the main change necessary to fix the issue, but it's not clear why all the other changes related to deletion vectors are part of this PR.

Reply:
I have now updated the PR description to explain why these changes were required.